app/models/name/create.rb

```diff
@@ -1,10 +1,15 @@
 # frozen_string_literal: true
 
 class Name < AbstractModel
+  scope :with_rank,
+        ->(rank) { where("`rank` = ?", Name.ranks[rank]) if rank }
+
   # Short-hand for calling Name.find_names with +fill_in_authors+ set to +true+.
   def self.find_names_filling_in_authors(in_str, rank = nil,
-                                         ignore_deprecated = false)
-    find_names(in_str, rank, ignore_deprecated, :fill_in_authors)
+                                         ignore_deprecated: false)
+    find_names(in_str, rank,
+               ignore_deprecated: ignore_deprecated,
+               fill_in_authors: true)
```

Comment:
How do you feel about having both positional and keyword parameters?

Reply:
Doesn't bother me.  In this specific case I switched to keywords due to a cop that prefers keywords over default values for booleans.  The real motivation for that doesn't apply in this case, but again I don't have a personal preference so I went with what the tool wants.